website/www/site/content/en/documentation/sdks/python-multi-language-pipelines-2.md

```diff
@@ -0,0 +1,216 @@
+
+Assuming you've built a JAR named **java-multilang-bundled-0.1.jar**, you can start the service with a command like the following, where `12345` is the port on which the expansion service will run:
+
+```
+java -jar java-multilang-bundled-0.1.jar 12345
+```
+
+For instructions on running an example expansion service, see [this README](https://github.com/apache/beam/blob/master/examples/multi-language/README.md#instructions-for-running-the-pipelines).
+
+## Create a Python pipeline
+
+Your Python pipeline can now use the [**ExternalTransformProvider**](https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.external_transform_provider.html#apache_beam.transforms.external_transform_provider.ExternalTransformProvider) API to configure your cross-language transform. Here’s an example constructed from [wordcount_external.py](https://github.com/apache/beam/blob/master/examples/multi-language/python/wordcount_external.py):
+
+First, determine the transform identifiers you are looking for:
+
+```python
+EXTRACT_IDENTIFIER = "beam:schematransform:org.apache.beam:extract_words:v1"
+COUNT_IDENTIFIER = "beam:schematransform:org.apache.beam:count:v1"
+WRITE_IDENTIFIER = "beam:schematransform:org.apache.beam:write_words:v1"
+```
+
+Then, initialize the `ExternalTransformProvider` with your expansion service. This can take two parameters:
+
+* `expansion_services`: an expansion service, or list of expansion services
+* `urn_pattern`: (optional) a regex pattern to match valid transforms
+
+```python
+provider = ExternalTransformProvider("localhost:" + expansion_service_port)
+```
+
+Next, retrieve each portable transform:
+```python
+Extract = provider.get_urn(EXTRACT_IDENTIFIER)
+Count = provider.get_urn(COUNT_IDENTIFIER)
+Write = provider.get_urn(WRITE_IDENTIFIER)
+```
+
+Finally, build your multi-language Python pipeline using a mix of native and portable transforms:
+```python
+with beam.Pipeline(options=pipeline_options) as p:
+_ = (p
+     | 'Read' >> beam.io.ReadFromText(input_path)
+     | 'Prepare Rows' >> beam.Map(lambda line: beam.Row(line=line))
+     | 'Extract Words' >> Extract(drop=["king", "palace"])
+     | 'Count Words' >> Count()
+     | 'Format Text' >> beam.Map(lambda row: beam.Row(line="%s: %s" % (
+          row.word, row.count))).with_output_types(
+          RowTypeConstraint.from_fields([('line', str)]))
+     | 'Write' >> Write(file_path_prefix=output_path))
+```
+
+## Run the pipeline
+
+The exact commands for running the Python pipeline will vary based on your environment. Assuming that your pipeline is coded in a file named **wordcount_external.py**, the steps should be similar to those below. For more information, see [the comments in addprefix.py](https://github.com/apache/beam/blob/41d585f82b10195f758d14e3a54076ea1f05aa75/examples/multi-language/python/addprefix.py#L18-L40).
+
+### Run with direct runner
+
+In the following command, `input1` is a file containing lines of text:
```

Comment:
Probably worth calling out that the expansion service needs to be started first (here and below in the Dataflow section)